CharlieBackend.Business/Services/AccountService.cs

```diff
@@ -170,5 +171,61 @@ public async Task<Result<AccountDto>> ChangePasswordAsync(ChangeCurrentPasswordD
 
             return Result<AccountDto>.GetError(ErrorCode.InternalServerError, "Salt for this account does not exist.");
         }
+
+        public async Task SendChangePasswordUrlAsync(string formUrl, ForgotPasswordDto forgotPassword)
```

Comment:
Your method is called SendChangePasswordUrl. Besides sending forgotPassword url, it also writes new token to account table. Let's call it GenerateForgotPasswordToken, it would be still non SRP, but at least will better illustrate what method does.